client/src/components/Map/Map.js

```diff
@@ -12,7 +12,7 @@ const Map = ({countryCode, countries, capitals, className, mapClassName}) => {
   const {language} = useContext(AppContext)
 
   const getCountriesCoordinates = () => {
-    countries.features.forEach(country => {
+    countries.features.find(country => {
```

Comment:
Not sure that there is much sense in replacing. I just advice you to think about some refactoring because you are mutating some out variables. Also if you don't put these variables in state, your ui won't change

Reply:
Speaking about that, I managed to change just one of two functions to make it pure, saving the result in State. As this component is loading a map 'on load', it need coordinates, but if to save coordinates in a state the map can't load and sends an errors message. Because setState is asynchronous. If I add a condition to load a map only if the coordinates are saved in a state then the whole document don't see variable 'map' . Maybe it would work if I declare 'let map;' variable before the condition, but again I would mutate the 'map' variable, so what the difference what variable  to  mutate?